vm/devices/user_driver/vfio_sys/src/lib.rs

```diff
@@ -186,6 +193,16 @@ impl Group {
         };
         Ok(GroupStatus::from(status.flags))
     }
+
+    pub fn set_keep_alive(&self, device_id: &str) -> anyhow::Result<()> {
+        // SAFETY: The file descriptor is valid and a correctly constructed struct is being passed.
+        unsafe {
+            let id = CString::new(device_id.to_owned())?;
+            ioctl::vfio_group_set_keep_alive(self.file.as_raw_fd(), id.as_ptr())
```

Comment:
Why did we put the keep alive IOCTL on the group instead of the device?

Reply:
Oh, I see, because it needs to be set before we create the device, is that right?

Reply:
Correct. A future rework for iommufd will eliminate this, but right now we need to send ioctl before the device created.